textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/plangen/OperatorGraph.java

```diff
@@ -0,0 +1,367 @@
+     * 
+     * PlanGenException is thrown if there is an operator not connected to the operator graph.
+     * 
+     */
+    private void checkGraphConnectivity() throws PlanGenException {
+        HashMap<String, HashSet<String>> undirectedAdjacencyList = new HashMap<>();
+        for (String vertexOrigin : adjacencyList.keySet()) {
+            undirectedAdjacencyList.put(vertexOrigin, new HashSet<>(adjacencyList.get(vertexOrigin)));
+        }
+        for (String vertexOrigin : adjacencyList.keySet()) {
+            for (String vertexDestination : adjacencyList.get(vertexOrigin)) {
+                undirectedAdjacencyList.get(vertexDestination).add(vertexOrigin);
+            }
+        }
+        
+        String vertex = undirectedAdjacencyList.keySet().iterator().next();
+        HashSet<String> unvisitedVertices = new HashSet<>(undirectedAdjacencyList.keySet());
+        
+        connectivityDfsVisit(vertex, undirectedAdjacencyList, unvisitedVertices);
+        
+        if (! unvisitedVertices.isEmpty()) {
+            throw new PlanGenException("Operators: " + unvisitedVertices + " are not connected to the operator graph.");
+        }   
+    }
+    
+    /*
+     * This is a helper function for checking connectivity by traversing the graph using DFS algorithm. 
+     */
+    private void connectivityDfsVisit(String vertex, HashMap<String, HashSet<String>> undirectedAdjacencyList, 
+            HashSet<String> unvisitedVertices) {
+        unvisitedVertices.remove(vertex);       
+        for (String adjacentVertex : undirectedAdjacencyList.get(vertex)) {
+            if (unvisitedVertices.contains(adjacentVertex)) {
+                connectivityDfsVisit(adjacentVertex, undirectedAdjacencyList, unvisitedVertices);
+            }
+        }
+    }
+    
+    /*
+     * Detects if there are any cycles in the operator graph.
+     * 
+     * This function uses a Depth First Search (DFS) algorithm to traverse the graph.
+     * It detects cycle by maintaining two lists of visited and visiting vertices,
+     * during the traversal, if it reaches an vertex that is in the visiting list, then there's a cycle.
+     * 
+     * PlanGenException is thrown if a cycle is detected in the graph.
+     * 
+     */
+    private void checkGraphCyclicity() throws PlanGenException {
+        HashSet<String> unvisitedVertices = new HashSet<>(adjacencyList.keySet());
+        HashSet<String> dfsPath = new HashSet<>();
+        
+        for (String vertex : adjacencyList.keySet()) {
+            if (dfsPath.contains(vertex)) {
+                checkCyclicityDfsVisit(vertex, unvisitedVertices, dfsPath);
+            }
+        }
+    }
+    
+    /*
+     * This is a helper function for detecting cycles by traversing the graph the graph using DFS algorithm. 
+     */
+    private void checkCyclicityDfsVisit(String vertex, HashSet<String> unvisitedVertices, 
```

Comment:
Codecov suggests that this function is not covered in test cases. I'll investigate on why this isn't covered and modify the test cases.